### DISCUSSION TITLE
Is there a way to upgrade my Unifi Controller without needing to destroy it?

### DISCUSSION THREAD

**Discussion by cofilippo** [tteck/Proxmox#2744] [Q&A]
The answers I have found so far seem to indicate that you basically need to build a new container and then restore a backup. I would like to know if there is not a more elegant way to get to a newer version of a controller.

**Comment by tteck** [comment]
The UniFi Network Server LXC script utilizes UniFis own APT repository to install the application. Updates are a simple `apt update && apt upgrade`. That's as "elegant" as it gets.

https://github.com/tteck/Proxmox/blob/932b69a8c47299048d49cb1167d7726091223008/install/unifi-install.sh#L38-L43

**Reply by MrDevanWright** [reply]
That's pretty elegant. Thank you for all the work you've put in.

Can this apt update/upgrade elegancy be used with all of your other scripts?

**Comment by Stephan-4711** [comment]
@tteck may I use this to ask, how the updates are working?
I used your script to install the unifi controller, when I used your script MongoDB4.4 was installed.
But the script never updated the DB to 7.0
Do I need to throw that away and reinstall it?

**Reply by Krissverse** [reply]
I'm with you here Stephan, did you ever find out how to upgrade properly from 4.4 to 7 to get the controller updated or did you just reinstall it?  I'm such a linux newb that I know it would be the easier option, but I still am hoping I can just upgrade the mongodb in the LXC without needing to start from scratch and restore from backup.

**Reply by tteck** [reply]
What's wrong with restoring a backup?

**Reply by tteck** [reply]
Remember the new UniFi Network Server requires a CPU with AVX Instruction Set.

**Reply by tteck** [reply]
To check your CPU, execute `lscpu | grep -o 'avx[^ ]*'` within the Proxmox shell, output should at least show `avx`

**Comment by tOOmki** [comment]
I've been using it for a few years.

https://community.ui.com/questions/UniFi-Installation-Scripts-or-UniFi-Easy-Update-Script-or-UniFi-Lets-Encrypt-or-UniFi-Easy-Encrypt-/ccbc7530-dd61-40a7-82ec-22b17f027776

**Reply by tteck** [reply]
Cumbersome

**Comment by VamperZ** [comment]
@tteck is correct, I was able to do an in-place update from Unifi 8.4.X to 8.5.X by using his script with an extra argument due to: "Repository 'https://dl.ui.com/unifi/debian stable InRelease' changed its 'Codename' value from 'unifi-8.4' to 'unifi-8.5'
N: This must be accepted explicitly before updates for this repository can be applied. See apt-secure(8) manpage for details."

Update took ~ 5 minutes without needing to rebuild anything. I'd still recommend a Unifi config backup just as a failsafe then if you encounter any issues, proceed with building a new Unifi LXC and restore from the backup you just took.

**sudo apt-get update --allow-releaseinfo-change && apt upgrade**
